addons/mail/models/discuss/discuss_channel_rtc_session.py

```diff
@@ -38,22 +45,46 @@ def create(self, vals_list):
         return rtc_sessions
 
     def unlink(self):
-        channels = self.channel_id
-        for channel in channels:
-            if channel.rtc_session_ids and len(channel.rtc_session_ids - self) == 0:
-                # If there is no member left in the RTC call, all invitations are cancelled.
-                # Note: invitation depends on field `rtc_inviting_session_id` so the cancel must be
-                # done before the delete to be able to know who was invited.
-                channel._rtc_cancel_invitations()
         notifications = [(channel, 'discuss.channel/rtc_sessions_update', {
             'id': channel.id,
             'rtcSessions': [('DELETE', [{'id': session_data['id']} for session_data in sessions_data])],
         }) for channel, sessions_data in self._mail_rtc_session_format_by_channel().items()]
+        session_ids_by_channel_by_url = defaultdict(lambda: defaultdict(list))
         for rtc_session in self:
             target = rtc_session.guest_id or rtc_session.partner_id
-            notifications.append((target, 'discuss.channel.rtc.session/ended', {'sessionId': rtc_session.id}))
-        self.env['bus.bus']._sendmany(notifications)
-        return super().unlink()
+            notifications.append((target, "discuss.channel.rtc.session/ended", {"sessionId": rtc_session.id}))
+            sfu_channel_uuid = rtc_session.channel_id.sfu_channel_uuid
+            url = rtc_session.channel_id.sfu_server_url
+            if not sfu_channel_uuid or not url:
+                continue
+            session_ids_by_channel_by_url[url][sfu_channel_uuid].append(rtc_session.id)
+        channels = self.channel_id
+        for channel in channels:
+            if channel.rtc_session_ids and len(channel.rtc_session_ids - self) == 0:
+                # If there is no member left in the RTC call, all invitations are cancelled.
+                # Note: invitation depends on field `rtc_inviting_session_id` so the cancel must be
+                # done before the unlink to be able to know who was invited.
+                channel._rtc_cancel_invitations()
```

Comment:
Can't you revert the diff on these lines? Ease of diff, but also ease to read the method as this doesn't interfere with the rest when it's at the top.

I think it's fine to loop multiple times on channel.